diskqueue.go

```diff
@@ -395,6 +436,184 @@ func (d *diskQueue) readOne() ([]byte, error) {
 	return readBuf, nil
 }
 
+func (d *diskQueue) removeBadFile(oldestBadFileInfo os.FileInfo) error {
+	var err error
+	badFileFilePath := path.Join(d.dataPath, oldestBadFileInfo.Name())
+
+	// remove file if it exists
+	err = os.Remove(badFileFilePath)
+	if err != nil {
+		d.logf(ERROR, "DISKQUEUE(%s) failed to remove .bad file(%s) - %s", d.name, oldestBadFileInfo.Name(), err)
+		d.updateTotalDiskSpaceUsed()
+		return err
+	} else {
+		// recaclulate total bad files disk size to get most accurate info
+		d.totalDiskSpaceUsed -= oldestBadFileInfo.Size()
+		d.logf(INFO, "DISKQUEUE(%s) removed .bad file(%s) of size(%d bytes) to free up disk space", d.name, oldestBadFileInfo.Name(), oldestBadFileInfo.Size())
+	}
+
+	return nil
+}
+
+func (d *diskQueue) removeReadFile() error {
+	var err error
+
+	if d.readFileNum == d.writeFileNum {
+		d.skipToNextRWFile()
+		return nil
+	}
+
+	if d.readFile == nil {
+		curFileName := d.fileName(d.readFileNum)
+
+		d.readFile, err = os.OpenFile(curFileName, os.O_RDONLY, 0600)
+		if err != nil {
+			return err
+		}
+
+		d.reader = bufio.NewReader(d.readFile)
+	}
+
+	closeReadFile := func() {
+		d.readFile.Close()
+		d.readFile = nil
+	}
+	defer closeReadFile()
+
+	// read total messages number at the end of the file
+	_, err = d.readFile.Seek(-numFileMsgBytes, 2)
+	if err != nil {
+		return err
+	}
+
+	var totalMessages int64
+	err = binary.Read(d.reader, binary.BigEndian, &totalMessages)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Extract this block into a function.
```go
    curFileName := d.fileName(d.readFileNum)
	totalMessages, err := d.readNumOfMessages(curFileName)
	if err != nil {
		return err
	}
```

```go
func (d *diskQueue) readNumOfMessages(fileName string) (int64, error) {
	var err error
	if d.readFile == nil {
		d.readFile, err = os.OpenFile(fileName, os.O_RDONLY, 0600)
		if err != nil {
			return 0, err
		}
		d.reader = bufio.NewReader(d.readFile)
	}

	closeReadFile := func() {
		d.readFile.Close()
		d.readFile = nil
	}
	defer closeReadFile()

	// read total messages number at the end of the file
	if _, err = d.readFile.Seek(-numFileMsgBytes, 2); err != nil {
		return 0, err
	}

	var totalMessages int64
	if err = binary.Read(d.reader, binary.BigEndian, &totalMessages); err != nil {
		return 0, err
	}
	return totalMessages, nil
}
```

Reply:
SG thank you for helping me with the readability of this code!